resolwe/flow/tests/test_serializers.py

```diff
@@ -148,6 +148,7 @@ def test_to_internal_value(self):
             'process': 'test-process',
             'descriptor_schema': 'test-schema',
         }
+        request.data = data
```

Comment:
This looks like an anti-pattern that you have to pass the same thing twice. You can try to add a read-only `collections` field to the serializer as values will be (probably) passed directly to the `validate` method, but won't affect the save method (you can also use `validate_collections` instead of `validate`).

If this is too complicated, you can leave it like that as it will get much simpler once the new relations will be merged.

Reply:
Agree. This was just an attempt to make tests happy. I would rather spare effort here and focus on porting m2m relations to m2one. Whean that is done, we can/should revisit this.